pkg/api/util/ipc_endpoint.go

```diff
@@ -0,0 +1,88 @@
+package util
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/system"
+)
+
+type IPCEndpoint struct {
+	client    *http.Client
+	target    url.URL
+	closeConn bool
+}
+
+func (end *IPCEndpoint) SetCloseConnection(state bool) {
+	end.closeConn = state
+}
```

Comment:
Is this function used?

Reply:
Not yet, but it will be needed by `cmd/agent/subcommands/diagnose/command.go`. It is now an Option, and is covered by the tests.